libpromises/evalfunction.c

```diff
@@ -238,7 +249,7 @@ static JsonElement* VarRefValueToJson(EvalContext *ctx, const FnCall *fp, const
             break;
 
         case RVAL_TYPE_CONTAINER:
-            convert = value;
+            convert = (JsonElement*) value;
```

Comment:
You const-away the constness of `value`. The compiler though still thinks that `value` is const, and may optimise things the way you are not expecting. And `value` is correct to be const, because you must not change the contents of the Variables table like that.

The easy solution here is to duplicate const to a second local variable. Cost: extra allocation.
If I was writing it from scratch I would have used 2 variables. One const variable containing the unmodifiable value from the variables table, and one non-const variable containing the allocated value.

In general never cast constness off. There are very few exceptions.